pwnlib/tubes/tube.py

```diff
@@ -420,13 +461,13 @@ def recvlines(self, numlines=2**20, keepends=False, timeout=default):
                 else:
                     break
 
-        if not keepends:
-            lines = [line.rstrip(self.newline) for line in lines]
+        if drop:
+            lines = [line[:-len(self.newline)] for line in lines]
```

Comment:
I hope something ensures self.newline is non-empty

Reply:
That's copy pasted from https://github.com/Gallopsled/pwntools/blob/f2f55f3e1c26e41e7b23f2fab0ff41baf7ebb27c/pwnlib/tubes/tube.py#L576
I think it's better than `.rstrip()`.

Reply:
> I hope something ensures self.newline is non-empty

Nothing checks if it's non-empty neither in tubes nor in context. Maybe it should though?

Reply:
Actually, recvline can return line without newline, so it should be rstrip.

https://github.com/Gallopsled/pwntools/blob/6f0793eebcf78eab4ac86bb8fcbb1ffee3e59fa6/pwnlib/tubes/tube.py#L531-L534